eva_submission/xlsx/xlsx_validation.py

```diff
@@ -145,3 +148,12 @@ def same_set(self, list1, list2, list1_desc, list2_desc):
             if list2_list1:
                 errors.append('%s present in %s not in %s' % (','.join(list2_list1), list2_desc, list1_desc))
             self.error_list.append('Check %s vs %s: %s' % (list1_desc, list2_desc, ' -- '.join(errors)))
+
+    def check_date(self, row, key, required=False):
+        if required and key not in row:
+            self.error_list.append(f'In row {row.get("row_num")}, {key} is required and missing')
+            return
+        if key in row and (isinstance(row[key], datetime) or str(row[key]).lower() in not_provided_check_list):
+            return
+        self.error_list.append(f'In row {row.get("row_num")}, {key} is not a date or "not provided": it is "{row[key]}"')
```

Comment:
If `required` is false you can still hit this line and have `key not in row`...

Also maybe I'm confused but I thought collection date was required now, so why does `required` default to false?

Reply:
The required field is still checked in the `complex_validation` function but I still changed the default to True.
I also changed the way the last error is reported to avoid getting `KeyError`